internal/fs/fs.go

```diff
@@ -197,6 +214,40 @@ func NewFileSystem(
 	return fs, nil
 }
 
+func createFileCacheHandler(cfg *ServerConfig) (fileCacheHandler *file.CacheHandler) {
+	var sizeInBytes uint64
+	// -1 means unlimited size for cache, the underlying LRU cache doesn't handle
+	// -1 explicitly, hence we pass MaxUint64 as capacity in that case.
+	if cfg.MountConfig.FileCacheConfig.MaxSizeInMB == -1 {
+		sizeInBytes = math.MaxUint64
+	} else {
+		sizeInBytes = uint64(cfg.MountConfig.FileCacheConfig.MaxSizeInMB) * util.MiB
+	}
+	fileInfoCache := lru.NewCache(sizeInBytes)
+
+	cacheLocation := string(cfg.MountConfig.CacheLocation)
+	// use temp-dir as default cache-location.
+	if cacheLocation == "" {
+		cacheLocation = cfg.TempDir
```

Comment:
What happens if cfg.TempDir is empty?

Reply:
Good catch, I thought cfg.TempDir will always be populated.  used os.TempDir if cfg.TempDir is empty.